src/utils/service/service.js

```diff
@@ -149,31 +150,75 @@ class Service {
    *
    * @method get
    * @param {Number} id - the ID of the resource.
-   * @param {Function} onSuccess - a callback to trigger on success
-   * @param {Function} onError - a callback to trigger on error
+   * @param {Object} options - an object with options for query params and success/error callbacks
+   * @param {Function} _onError - [Deprecated] a callback to trigger on error
    * @return {Void}
    */
-  get = (id, onSuccess, onError) => {
-    this.client.get(String(id)).then(
-      this.handleResponse.bind(this, onSuccess),
-      this.handleResponse.bind(this, onError)
-    );
+
+  get = (id, options, _onError) => {
+    let request;
+
+    if (typeof (options) === 'function') {
+      Logger.deprecate('Passing onSuccess and onError as separate arguments is deprecated. Please use `get(1, { onSuccess: function, onError: function, params { k: "v" } })`'); // eslint-disable-line max-len
+
+      request = this.client.get(String(id));
+
+      request.then(
+        this.handleResponse.bind(this, options),
+        this.handleResponse.bind(this, _onError)
+      );
+    } else {
+      if (typeof (options.params) === 'undefined') {
+        options.params = {};
+      }
+
+      request = this.client.get(String(id),
```

Comment:
this gives developers flexibility to add `params`, but what if they want to pass something additional supported by axios? How about instead of just passing an object with `params` that you remove `onSuccess` and `onError` and send whatevers left? allowing devs to pass a larger config object?